src/java.desktop/share/classes/sun/awt/image/GifImageDecoder.java

```diff
@@ -395,7 +395,16 @@ private int sendPixels(int x, int y, int width, int height,
         int off = y * global_width + x2;
         boolean save = (curframe.disposal_method == GifFrame.DISPOSAL_SAVE);
         if (trans_pixel >= 0 && !curframe.initialframe) {
-            if (saved_image != null && model.equals(saved_model)) {
+            boolean isSimple = saved_image != null && model.equals(saved_model);
+            if (isSimple) {
+                for (int a = 0; a < saved_image.length; a++) {
+                    if ((saved_image[a] & 0xff) == trans_pixel) {
```

Comment:
This check is scanning whole saved image(until 241 index in this scenario) for each scanline. So for each GIF frame we are running this loop height times.

We can maintain a common boolean variable and verify isSimple/not just before parseImage() call. This will make sure that we perform this check only once per GIF frame.

To verify, updated the code to move this check inside readImage() and just before parseImage() call and it works fine.

Reply:
Good point. I [added the field](https://github.com/openjdk/jdk/pull/25264/commits/b8a8bf7d3d81372f686d48aec2a1733955ebd545) `isSimpleSavedImageComparison`. In addition to the unit test I also tested the gifs mentioned in the PR description.